test/specs/adapter.spec.js

```diff
@@ -1,7 +1,8 @@
 var axios = require('../../index');
 
 describe('adapter', function () {
-  it('should support custom adapter', function (done) {
+  it('should support custom adapter (asynchronously)', function (done) {
+    var promiseResolveSpy = spyOn(window.Promise, 'resolve').and.callThrough();
```

Comment:
Ping for this one.

Reply:
Ah, I forgot to change all the occurrences. Did that now.

Reply:
You checked the flag in interceptors now. The ideal position is in the adapter, but I also understand that we don't have a hook there unless using a customized adapter. Do you have better solutions?

Reply:
A better solution on how to unit test whether or not the interceptor code is executed synchronously? I think your idea of the flag is a good pattern, I'm not sure where else to put it except for inside the interceptor. The other option is listening for the promise.

Reply:
In fact, we want to confirm not only the interceptor code is executed synchronously, but also the adapter. Because the request is send by the adapter.

You can add 2 simple tests in `test/specs/adapter.spec.js`. One has no interceptors(which should be sync) and the other one has an interceptor(without `synchronous` set, which means async).

Reply:
Good call, I've added the unit tests.